OCI: Learn about claims concept page (#5670)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/concepts/oauth-claims/index.md

---
title: OAuth 2.0 and OpenID Connect claims
meta:
  - name: description

>     content: A high-level overview of OAuth 2.0 and OpenID Connectclaims.

Reviewer: ```suggestion
    content: A high-level overview of OAuth 2.0 and OpenID Connect claims.
```

---

# Learn about OAuth 2.0 and OpenID Connect claims

[OAuth 2.0 and OpenID Connect (OIDC)](/docs/concepts/oauth-openid) claims are key-value pairs of data that contain information, typically information about a user. Okta uses these claims to provide context about the user and their permissions.

> This page provides an overview of OAuth 2.0 and OIDC claims, including their types and usage, and how they're used in Okta.

Reviewer: ```suggestion
This page provides an overview of OAuth 2.0 and OIDC claims, including their types and how they're used in Okta.
```

---

## What are claims

> At a basic level, a claim is a piece of information asserted about a subject. In the context of OAuth 2.0 and OpenID Connect (OIDC), claims are key-value pairs of data contained within tokens. Claims are commonly packaged into [access](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#access-token) and [ID tokens](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#id-token). Access tokens and ID tokens are formatted as JSON Web Tokens (JWTs).

Reviewer 2: ```suggestion
At a basic level, a claim is a piece of information that's asserted about a subject. In the context of OAuth 2.0 and OpenID Connect (OIDC), claims are key-value pairs of data contained within tokens. Claims are commonly packaged into [access](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#access-token) and [ID tokens](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#id-token). Access tokens and ID tokens are formatted as JSON Web Tokens (JWTs).
```

Access and ID tokens contain a [header](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#access-token-header), [payload](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#access-token-payload), and [signature](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#access-token-signature).

Along with other parameters, the payload can contain scopes and custom claims. See the following example of an access token payload.

```json
{
  "ver": 1,
  "jti": "AT.0mP4JKAZX1iACIT4vbEDF7LpvDVjxypPMf0D7uX39RE",
  "iss": "https://{yourOktaDomain}/oauth2/{authorizationServerId}",
  "aud": "https://api.example.com",
  "sub": "00ujmkLgagxeRrAg20g3",
  "iat": 1467145094,
  "exp": 1467148694,
  "cid": "nmdP1fcyvdVO11AL7ECm",
  "uid": "00ujmkLgagxeRrAg20g3",
  "scp": [
    "openid",
    "email",
  ],
  "auth_time": 1467142021,
}
```

> The payload contains various [reserved claims](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#reserved-claims-in-the-payload-section), along with an array of scopes, within the `scp` array. The reserved claims convey some information about the access token. For example, the `iss` (issuer) claim contains the URL of the authorization server that issued the token. And the `aud` (audience) claim contains the intended recipient of the token.

Reviewer: Might want to add a sentence here that you're talking about the example above, like "in the access token example, the payload contains....."

Because when I got to the next paragraph, you said the `scp` parameter passes two scopes to the resource server:  but it can pass more than these two. It could be misleading to newbies.

See the following example of an ID token payload.

```json
{
  "sub": "user123",
  "email": "user@example.com",
  "phone_number": "+1234567890",
  "name": "John Doe"
}
```

In this example, the ID token payload passes the user ID (`sub`), user's email (`email`), phone number (`phone_number`), and their name (`name`). These claims identify the user to the client app.

See [ID token](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/overview/#id-token) for information about how ID tokens are structured and used in Okta.

## Claim types

Claims can be categorized into two groups based on how they're defined and used.

**Registered claims**: There are seven registered claims that are defined in the [JWT spec](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1). They provide fundamental information like the token's issuer (`iss`), the user's unique ID (`sub`), or when the token expires (`exp`).

> **Custom claims**: Custom claims consist of [public claims](#public-claims) that third-parties have registered and [private claims](#private-claims) that are created for use between specific parties.

Reviewer: ```suggestion
**Custom claims**: Custom claims consist of [public claims](#public-claims) that third parties have registered and [private claims](#private-claims) that are created for use between specific parties.
```

### Public claims

> Public claims are designed to avoid naming collision with other registered claims. Public claims include the [OIDC standard claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims). For a full list of public claims, see the [Internet Assigned Numbers Authority (IANA) registry](https://www.iana.org/assignments/jwt/jwt.xhtml).

Reviewer: ```suggestion
Public claims are designed to avoid naming collisions with other registered claims. Public claims include the [OIDC standard claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims). For a full list of public claims, see the [Internet Assigned Numbers Authority (IANA) registry](https://www.iana.org/assignments/jwt/jwt.xhtml).
```

## Different ways to implement claims

You can use claims in different ways with your Okta org.

* [Custom claims](/docs/guides/configure-custom-claims/oktaoidc/main/): Learn how to customize the claims that are returned in Okta access and ID tokens. You can add custom claims to access tokens, or add custom claims to ID tokens that are tailored for specific user traits, based on your use case.
* [Custom group claims](/docs/guides/customize-tokens-returned-from-okta/main/): Similar to custom claims, you can add a custom groups claim to access and ID tokens. You can customize the tokens based on app and user groups.
* [Federated claims](/docs/guides/configure-federated-claims/oktaoidc/main/): If you have OIDC or SAML apps, learn how to create app entitlement claims that allow you to control permissions for users within specific apps.
* [Claims sharing](/docs/guides/configure-claims-sharing/oktaoidc/main/): If you use an OIDC or SAML 2.0 identity provider (IdP), learn how to share identity-related information (claims) between different orgs to enable secure access to resources.

### Next steps

> * [Learn about tokens](/docs/concepts/token-lifecycles/#token-types) and then learn [how to implement them](/docs/guides/tokens/)

Reviewer: ```suggestion
* [Learn about tokens](/docs/concepts/token-lifecycles/#token-types) and then learn [how to implement them](/docs/guides/tokens/).
```

> * Explore [JSON Web Token (JWT) claims](https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-token-claims)

Reviewer: ```suggestion
* Explore [JSON Web Token (JWT) claims](https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-token-claims).
```

> * Review [sample use cases for scopes and claims](https://auth0.com/docs/get-started/apis/scopes/sample-use-cases-scopes-and-claims)

Reviewer: ```suggestion
* Review [sample use cases for scopes and claims](https://auth0.com/docs/get-started/apis/scopes/sample-use-cases-scopes-and-claims).
```
